api/http/api_inventory.go

```diff
@@ -984,6 +986,66 @@ func (i *inventoryHandlers) GetDeviceGroupsInternalHandler(w rest.ResponseWriter
 	w.WriteJson(res)
 }
 
+func (i *inventoryHandlers) ReindexDeviceDataHandler(w rest.ResponseWriter, r *rest.Request) {
+	ctx := r.Context()
+	tenantId := r.PathParam("tenant_id")
+	ctx = getTenantContext(ctx, tenantId)
+
+	l := log.FromContext(ctx)
+
+	deviceId := r.PathParam("device_id")
+	if len(deviceId) < 1 {
+		u.RestErrWithLog(w, r, l, errors.New("device id cannot be empty"), http.StatusBadRequest)
+		return
+	}
+
+	serviceName, err := utils.ParseQueryParmStr(r, "service", false, nil)
+	// inventory service accepts only reindex requests from devicemonitor
+	if err != nil || serviceName != "devicemonitor" {
```

Comment:
nit: could we have an array `supportedServices`?

Reply:
I don't feel a need for it; this is only a workaround for not having the reporting in the release so I think we'll have very nice implementation in the reporting service and having more rough one here should be OK